### PR TITLE
Add floyd warshall 2

### PR DESCRIPTION
**Pull request by mjh316**
Added floyd-warshall algorithm.

### CODE REVIEW

**Comment by er888kh** [comment]
> Is it okay if this is reviewed as soon as possible? I fear the more merge conflicts there become, the more maintenance on the DIRECTORY.MD that'll have to be done.

Don't worry, fixing those is trivial and can be done just before the merge by maintainer.

**Review by siriak — APPROVED** [review]
Looks good, thanks! Sorry that it takes so long to get it merged, I just have too many PRs to review :(

**Pull request merged**
